src/main/java/com/minecolonies/coremod/colony/requestsystem/requests/StandardRequestFactories.java

```diff
@@ -0,0 +1,304 @@
+            for (final IToken token : request.getChildren())
+            {
+                childrenCompound.appendTag(controller.serialize(token));
+            }
+
+            compound.setTag(NBT_TOKEN, tokenCompound);
+            compound.setTag(NBT_STATE, stateCompound);
+            compound.setTag(NBT_REQUESTED, requestedCompound);
+
+            if (request.hasResult())
+            {
+                compound.setTag(NBT_RESULT, request.getResult().serializeNBT());
+            }
+
+            if (request.hasParent())
+            {
+                compound.setTag(NBT_PARENT, controller.serialize(request.getParent()));
+            }
+
+            compound.setTag(NBT_CHILDREN, childrenCompound);
+
+            return compound;
+        }
+
+        /**
+         * Method to deserialize a given Request.
+         *
+         * @param controller The controller that can be used to deserialize complicated types.
+         * @param nbt        The data of the request that should be deserialized.
+         * @return The request that corresponds with the given data in the nbt
+         */
+        @NotNull
+        @Override
+        @SuppressWarnings("squid:LeftCurlyBraceStartLineCheck")
+        /**
+         * Moving the curly braces really makes the code hard to read.
+         */
+        public StandardRequests.ItemStackRequest deserialize(@NotNull final IFactoryController controller, @NotNull final NBTTagCompound nbt)
+        {
+            final IToken token = controller.deserialize(nbt.getCompoundTag(NBT_TOKEN));
+            final RequestState state = RequestState.deserializeNBT((NBTTagInt) nbt.getTag(NBT_STATE));
+            final ItemStack requested = new ItemStack(nbt.getCompoundTag(NBT_REQUESTED));
+
+            final List<IToken> childTokens = new ArrayList<>();
+            final NBTTagList childCompound = nbt.getTagList(NBT_CHILDREN, Constants.NBT.TAG_COMPOUND);
+            for (int i = 0; i < childCompound.tagCount(); i++)
+            {
+                childTokens.add(controller.deserialize(childCompound.getCompoundTagAt(i)));
+            }
+
+            final StandardRequests.ItemStackRequest request = controller.getNewInstance(requested, new TypeToken<StandardRequests.ItemStackRequest>() {}, token, state);
+
+            if (nbt.hasKey(NBT_PARENT))
+            {
+                request.setParent(controller.deserialize(nbt.getCompoundTag(NBT_PARENT)));
+            }
+
+            if (nbt.hasKey(NBT_RESULT))
+            {
+                request.setResult(new ItemStack(nbt.getCompoundTag(NBT_RESULT)));
+            }
+
+            return request;
+        }
+    }
+
+    @SuppressWarnings("squid:S2972")
+    /**
+     * We have this class the way it is for a reason.
+     */
+    public static final class DeliveryFactory implements IRequestFactory<Delivery, StandardRequests.DeliveryRequest>
+    {
+
+        @NotNull
+        @Override
+        @SuppressWarnings("squid:LeftCurlyBraceStartLineCheck")
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Suppressing the 'squid:LeftCurlyBraceStartLineCheck' warning is not allowed [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1309)